src/main/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -187,7 +212,7 @@ private BAMFileReader(final BlockCompressedInputStream compressedInputStream,
                           final SAMRecordFactory factory)
         throws IOException {
         mIndexStream = indexStream;
-        mIsSeekable = true;
+        mIsSeekable = null != indexStream;
```

Comment:
I think you're right and I can leave it to "true". I ran SamReaderFactoryTest and all 84 tests pass.